drivers/spi/spi_nrfx_spi.c

```diff
@@ -317,9 +344,24 @@ static int spi_nrfx_pm_action(const struct device *dev,
 			? NRF_GPIO_PIN_PULLDOWN		\
 			: NRF_GPIO_PIN_NOPULL)
 
+#define SPI_NRFX_SPI_PIN_CFG(idx)			\
+	COND_CODE_1(CONFIG_PINCTRL,			\
+		(.skip_gpio_cfg = true,			\
+		 .skip_psel_cfg = true,),		\
+		(.sck_pin   = SPI_PROP(idx, sck_pin),	\
+		 .mosi_pin  = SPI_PROP(idx, mosi_pin),	\
+		 .miso_pin  = SPI_PROP(idx, miso_pin),	\
+		 .miso_pull = SPI_NRFX_MISO_PULL(idx),))
+
 #define SPI_NRFX_SPI_DEVICE(idx)					       \
-	BUILD_ASSERT(							       \
-		!SPI_PROP(idx, miso_pull_up) || !SPI_PROP(idx, miso_pull_down),\
+	BUILD_ASSERT((IS_ENABLED(CONFIG_PINCTRL) &&			       \
+		      DT_PINCTRL_HAS_IDX(SPI(idx), 0)) ||		       \
+		     (!IS_ENABLED(CONFIG_PINCTRL) &&			       \
+		      DT_NODE_HAS_PROP(SPI(idx), sck_pin)),		       \
+		"No pins assigned to SPI"#idx);				       \
```

Comment:
ditto, sleep case should also be checked. I think creating an intermediate helper macro would improve readability here

Reply:
I'll add a generic macro for checking this, to not copy almost the same code in several drivers.

Reply:
Added.